Python/Product/Workspace/PythonLaunchDebugTargetProvider.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using Microsoft.VisualStudio.Workspace;
+using Microsoft.VisualStudio.Workspace.Debug;
+
+namespace Microsoft.PythonTools.Workspace {
+    [ExportLaunchConfigurationProvider(
+        ProviderType,
+        new[] { PythonConstants.FileExtension, PythonConstants.WindowsFileExtension },
+        PythonDebugLaunchProvider.LaunchTypeName,
+        PythonDebugLaunchProvider.JsonSchema
+    )]
+    class PythonLaunchDebugTargetProvider : ILaunchConfigurationProvider {
```

Comment:
>PythonLaunchDebugTargetProvider [](start = 10, length = 31)

You might want to base the name of the class on the interface it is implementing.  Right now it's confusing because there's a class that implements ILaunchDebugTargetProvider, and it's not this one.  So maybe "PythonLaunchDebugTargetProvider" for the other class, and "PythonLaunchConfigurationProvider" for this class.  Same comment for the Pyproj ones.